server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryWebAdminServerIntegrationTest.java

```diff
@@ -23,16 +23,19 @@
 import org.apache.james.JamesServerBuilder;
 import org.apache.james.JamesServerExtension;
 import org.apache.james.MemoryJamesServerMain;
-import org.apache.james.webadmin.integration.WebAdminServerIntegrationTest;
+import org.apache.james.probe.DataProbe;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.webadmin.integration.WebAdminServerIntegrationContract;
 import org.apache.james.webadmin.integration.WebadminIntegrationTestModule;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-class MemoryWebAdminServerIntegrationTest extends WebAdminServerIntegrationTest {
+class MemoryWebAdminServerIntegrationTest implements WebAdminServerIntegrationContract {
 
     @RegisterExtension
     static JamesServerExtension jamesServerExtension = new JamesServerBuilder()
         .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
             .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
             .overrideWith(new WebadminIntegrationTestModule()))
+        .resolveParam(DataProbe.class, jamesServer -> jamesServer.getProbe(DataProbeImpl.class))
```

Comment:
Mmmh I wonder if we can not make **probe** registration implicit:

Given:

```
@Test
void test(DataProbeImpl probe) { ... } 
```

The JamesServerExtension knows that it should be calling `jamesServer.getProbe(DataProbeImpl.class)` to be resolving the parameter: we can make this slightly more fluent IMO...

Reply:
We could introduce a `Probe` interface to tag every probes and have a `ProbeParamResolver` that would look at a way to resolve any probe type parameter with a guice Injector

Reply:
No need of new type. GuiceJamesServer always return GuiceProbe instances